dolphinscheduler-task-plugin/dolphinscheduler-task-api/src/main/java/org/apache/dolphinscheduler/plugin/task/api/AbstractYarnTask.java

```diff
@@ -69,17 +81,77 @@ public void handle() throws TaskException {
         }
     }
 
+    // todo
+    @Override
+    public void submitApplication() throws TaskException {
+
+    }
+
+    // todo
+    @Override
+    public void trackApplicationStatus() throws TaskException {
+
+    }
+
     /**
      * cancel application
      *
-     * @param status status
-     * @throws Exception exception
+     * @throws TaskException exception
      */
     @Override
-    public void cancelApplication(boolean status) throws Exception {
-        cancel = true;
+    public void cancelApplication() throws TaskException {
         // cancel process
-        shellCommandExecutor.cancelApplication();
+        try {
+            shellCommandExecutor.cancelApplication();
+        } catch (Exception e) {
+            throw new TaskException("cancel application error", e);
+        }
+    }
+
+    /**
+     * get application ids
+     * @return
+     * @throws TaskException
+     */
+    public Set<String> getApplicationIds() throws TaskException {
```

Comment:
Here exist a utils can directly get the appIds.
```
org.apache.dolphinscheduler.plugin.task.api.utils.LogUtils.getAppIdsFromLogFile(@NonNull String logPath)
```

Reply:
done